packages/redis/src/RedisClient.ts

```diff
@@ -380,62 +353,68 @@ export class RedisClient implements IRedisClient, IRequireInitialization, IDispo
         }
     }
 
-    public async xReadGroupObject<T>(
+    public async xReadGroup(
         context: SpanContext,
-        type: string | IClassType<T>,
         streamName: string,
         consumerGroup: string,
         consumerName: string,
+        count: number,
         id: string = ">"
-    ): Promise<[string, T]> {
+    ): Promise<IRedisMessage[]> {
         const db = this.config.db;
         const span = this.tracer.startSpan("Redis Client xReadGroupObject Call", {
             childOf: context,
         });
-        this.spanLogAndSetTags(span, this.xReadGroupObject.name, this.config.db, id, streamName);
+
+        this.spanLogAndSetTags(span, this.xReadGroup.name, this.config.db, id, streamName);
+
         try {
-            const typeName = this.getTypeName(type);
             const response = await this.client.xreadgroup([
                 "group",
                 consumerGroup,
                 consumerName,
+                "count",
+                String(count),
                 "block",
                 "0",
                 "streams",
                 streamName,
                 id,
             ]);
-            const value = extractStreamValue(response);
-            const streamId = extractStreamId(response);
-            let data;
-            if (value) {
+
+            console.log("HERE YO: ", response);
+
+            const results = extractStreamValues(response);
+
+            const messages: IRedisMessage[] = results.map(({ streamId, data, type }) => {
                 const buf = this.config.base64Encode
-                    ? Buffer.from(value, "base64")
-                    : Buffer.from(value);
-                const msg = this.encoder.decode(new Uint8Array(buf), typeName);
-                data = msg.payload;
-            }
+                    ? Buffer.from(data, "base64")
+                    : Buffer.from(data);
+
+                return { streamId, ...this.encoder.decode(new Uint8Array(buf), type) };
```

Comment:
What will happen here if the payload contains a field named `streamId`? 🤔

Reply:
i think this works because `decode` returns `IMessage` and the return value would be `{streamId: string; payload: any; type: string;}`

Reply:
Ah, of course. Never mind. 😄